griptape/tasks/prompt_task.py

```diff
@@ -203,21 +203,24 @@ def try_run(self) -> BaseArtifact:
             self.prompt_driver.tokenizer.stop_sequences.extend([self.response_stop_sequence])
 
         result = self.prompt_driver.run(self.prompt_stack)
-        subtask = self.add_subtask(ActionsSubtask(result.to_artifact()))
-
-        while True:
-            if subtask.output is None:
-                if len(self.subtasks) >= self.max_subtasks:
-                    subtask.output = ErrorArtifact(f"Exceeded tool limit of {self.max_subtasks} subtasks per task")
+        if self.tools:
+            subtask = self.add_subtask(ActionsSubtask(result.to_artifact()))
+
+            while True:
```

Comment:
You could avoid some nesting if you flip the condition:

```py
if not self.tools:
    self.output = result.to_artifact()
    return self.output

while True:
    ...
```

Reply:
I'd rather keep the conditional positive and solve the ugly nested code separately [here](https://github.com/griptape-ai/griptape/issues/1472).